helper/schema/schema_test.go

```diff
@@ -3595,6 +3595,244 @@ func TestSchemaMap_InternalValidate(t *testing.T) {
 			true,
 		},
 
+		"ConflictsWith list index syntax with self reference": {
+			map[string]*Schema{
+				"config_block_attr": &Schema{
+					Type:     TypeList,
+					Optional: true,
+					Elem: &Resource{
+						Schema: map[string]*Schema{
+							"nested_attr": &Schema{
+								Type:          TypeString,
+								Optional:      true,
+								ConflictsWith: []string{"config_block_attr.0.nested_attr"},
+							},
+						},
+					},
+				},
+			},
+			false, // TODO: This should return an error as it will always error during runtime
```

Comment:
Can you let me know more details on this TODO? If the test is supposed to error but currently doesn't doesn't that mean something is broken?

Reply:
Hi @appilon 👋 Yes, that TODO is meant to signify something is broken in the sense that `InternalValidate()` does not fail when you self-reference in this manner, however the schema is invalid and will never work when the attribute is declared in a configuration.

Copying this configuration block schema into a real attribute of a real resource (I just randomly chose `aws_devicefarm_project`):

```go
func resourceAwsDevicefarmProject() *schema.Resource {
	return &schema.Resource{
		Create: resourceAwsDevicefarmProjectCreate,
		Read:   resourceAwsDevicefarmProjectRead,
		Update: resourceAwsDevicefarmProjectUpdate,
		Delete: resourceAwsDevicefarmProjectDelete,

		Schema: map[string]*schema.Schema{
			"arn": {
				Type:     schema.TypeString,
				Computed: true,
			},

			"name": {
				Type:     schema.TypeString,
				Required: true,
			},

			"config_block_attr": {
				Type:     schema.TypeList,
				Optional: true,
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"nested_attr": {
							Type:          schema.TypeString,
							Optional:      true,
							ConflictsWith: []string{"config_block_attr.0.nested_attr"},
						},
					},
				},
			},
		},
	}
}
```

And adding the configuration to the test configuration:

```go
func testAccDeviceFarmProjectConfig(rInt int) string {
	return fmt.Sprintf(`
resource "aws_devicefarm_project" "foo" {
  name = "tf-testproject-%d"

  config_block_attr {
    nested_attr = "test"
  }
}
`, rInt)
}
```

Will always return a runtime error due to the self-reference:

```
=== CONT  TestAccAWSDeviceFarmProject_basic
    TestAccAWSDeviceFarmProject_basic: testing.go:669: Step 0 error: config is invalid: "config_block_attr.0.nested_attr": conflicts with config_block_attr.0.nested_attr
```